rs/utils/src/str.rs

```diff
@@ -93,7 +93,9 @@ impl StrEllipsize for str {
         }
 
         const ELLIPSIS: &str = "...";
-        assert!(max_len >= ELLIPSIS.len());
+        if max_len <= ELLIPSIS.len() {
```

Comment:
Maybe a more correct version would be to use a sub-string of ellipsis just in case someone changes the text inside:

```rust
const ELLIPSIS: &str = "...";
if max_len <= ELLIPSIS.len() {
    return ELLIPSIS[..max_len].to_string();
}
```

Reply:
This would be dangerous because if `max_len` doesn't point to a char boundary it would [panic](https://doc.rust-lang.org/std/primitive.str.html#impl-SliceIndex%3Cstr%3E-for-(Bound%3Cusize%3E,+Bound%3Cusize%3E)) (this is also the reason for why the `safe_truncate` functions above exist). 

At this point I'm wondering whether it wouldn't be better to simply go for returning an empty string whenever `max_len < ELLIPSIS.len()`. It is trivially correct and there is probably no real usecase for `max_len < ELLIPSIS.len()` anyways -- the only thing we care about in such a case is that it doesn't crash.

WDYT?

Reply:
ok, empty line is fine with me too, it has no real usage, so no need for fancy handling, you're right.

Reply:
Ok, moved back to empty string in 3236679